server/controller/user.js

```diff
@@ -37,72 +37,48 @@ class UserController{
     }
 
     static signInUser (req, res, next){
+        let email = req.body.email.split(' ').join('');
         const text = "SELECT * FROM users WHERE email = $1";
-        const values = [req.body.email]
-        // console.log("meme")
+        const values = [email]
        client.query(text, values, (err, data) => {
-           if(data) {
-               if(!data){
-                   return res.status(404).json({
-                       success: false,
-                       message: 'Authentication failed. User not found'
-                   })
-               }else if(data){
-                   //console.log(data.rows.password)
-                   let dataPassword = ''
-                   let dataEmail = ''
-                   let dataUserId = 0;
-                   
-                   //console.log(data.rows)
-                   data.rows.map(user => {
-                       //console.log(user)
-                       //console.log(user.id)
-                     dataUserId += user.id;
-                      dataEmail += user.email;
-                      //console.log(datauserIdNum)
-
-                   })
-                   
-                   //console.log(dataUserId)
-                   data.rows.map((user) => {
-                       if (req.body.password == user.password){
-                           dataPassword += user.password
-                       }
-                   });
-                   if(dataPassword !== req.body.password){
-                       //console.log(dataPassword)
-                       res.status(400).json({
-                           success: false,
-                           message: 'Authentication failed, wrong Password'
-                       });
-                   }else{
-                       bcrypt.compare(req.body.password, dataPassword, (err, response) => {
-                           if(err){
-                               return res.status(500).json({
-                                   error: err.message
-                               })
-                           }else{
+            if(!(data.rows[0])){
+                return res.status(404).json({
+                    success: false,
+                    message: 'User does not exist.'
+                })
+            }else if(data.rows[0]){
+                if(data.rows[0].password !== req.body.password){
```

Comment:
You don't need this check at all because the password would never be the same in any case. You are comparing what is not encrypted with what has not been encrypted, it is wrong.

Reply:
fixed.